### PR TITLE
Feat: SpringMotionPanel with SpringMotionWindowProtocol conformance 

### PR DESCRIPTION
**Pull request by ostenemes**
Added `SpringMotionWindowProtocol` and `SpringMotionBase` to enable window pinning functionality for both `NSWindow` and `NSPanel` classes.

### CODE REVIEW

File: Sources/SpringMotionWindowProtocol.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+#if canImport(AppKit)
+import AppKit
+
+public protocol SpringMotionWindowProtocol: NSWindow {
+    var configuration: SpringConfiguration { get set }
+    var destinationPoint: NSPoint? { get set }
+    var motionPhysics: SpringMotionPhysics { get set }
+    var currentMotionState: SpringMotionState? { get set }
+    var displayLink: CVDisplayLink? { get set }
+    var anchorWindowFrameObservation: NSKeyValueObservation? { get set }
```
Comment (anton-barkov) [inline Sources/SpringMotionWindowProtocol.swift]: I'm sorry, but I don't think that making `displayLink` and the rest of these properties public is a good idea. I'd prefer they be left "under the hood" — clients of these UI components shouldn't need to read or modify them.

I understand that you're trying to minimize code duplication. I think that in our case, the implementation of `SpringMotionPanel` via duplication of `SpringMotionWindow` is the way to go. At least, I can't see any other way that doesn't involve exposing the technical innards of these components.

**Review by anton-barkov — CHANGES_REQUESTED** [review]
Thanks for the addition! 
I suggest we redo this in a slightly more code-smelly but end-user-friendly manner.

File: Sources/SpringMotionWindowProtocol.swift (hunk 1 above)
Comment (ostenemes) [reply Sources/SpringMotionWindowProtocol.swift]: That was my first intention and frirst cmmit 😅 alright, reverted last changes.

Comment (anton-barkov) [reply Sources/SpringMotionWindowProtocol.swift]: Nice, thanks :)
Much easier to read now as well, IMO.

**Review by anton-barkov — DISMISSED** [review]

**Review by anton-barkov — APPROVED** [review]

**Pull request merged**